internal/clientgen/testdata/expected_baseauth_typescript.ts

```diff
@@ -72,6 +72,9 @@ export interface ClientOptions {
      */
     fetcher?: Fetcher
 
+	/** indicates whether the user agent should send or receive cookies from the other domain in the case of cross-origin requests. */
+	credentials?: RequestCredentials;
```

Comment:
Having thought about this some more, I think it would be better if we had a field `request?: RequestInit` instead, and we merge those options with the per-request data. That seems more flexible and would support other `RequestInit` fields at the same time, without needing to copy each of those fields to `ClientOptions` over time. What do you think?

Reply:
Make sense, let me try to update it